assembly/src/tests.rs

```diff
@@ -1327,24 +1336,21 @@ fn program_with_nested_procedure() -> TestResult {
     let program = context.assemble(source)?;
     let expected = "\
 begin
-    join
-        join
-            join
-                basic_block push(2) push(4) add end
-                basic_block push(3) push(7) mul end
-            end
-            join
-                basic_block push(11) end
-                join
-                    join
-                        basic_block push(5) end
-                        basic_block push(3) push(7) mul end
-                    end
-                    basic_block add end
-                end
-            end
-        end
-        basic_block neg add end
+    basic_block
+        push(2)
+        push(4)
+        add
+        push(3)
+        push(7)
+        mul
+        push(11)
+        push(5)
+        push(3)
+        push(7)
+        mul
+        add
+        neg
+        add
     end
 end";
```

Comment:
Ah - interesting! It seems like we are inlining simple procedures, right?

Reply:
Right - execs [return the root](https://github.com/0xPolygonMiden/miden-vm/blob/c00fef2615e889cbec7b0b43f6e67aedb81741b4/assembly/src/assembler/instruction/procedures.rs#L88) of the procedure directly, and so it ends up getting inlined

Reply:
Just to be clear - this is only inlining `exec` when the callee will fit in the current `span`? Or is this inlining all `exec`s?

Reply:
My guess is that it is inlining all procedures where a procedure is a single basic block.

Reply:
> My guess is that it is inlining all procedures where a procedure is a single basic block.

This is my understanding as well. In the case of execs, `invoke_mast_root()` returns the procedure root's `MastNodeId`. If this `MastNodeId` refers to a basic block, then it will get merged with surrounding basic blocks.